python/triton/runtime/autotuner.py

```diff
@@ -101,6 +102,8 @@ def run(self, *args, **kwargs):
                 self.cache[key] = builtins.min(timings, key=timings.get)
                 self.hook(args)
                 self.configs_timings = timings
+                if self.verbose:
+                    print(str(key) + ": " + str(self.cache[key]))
```

Comment:
Should we move this print after line 109?

Reply:
We only want to print out the best config once, i.e. when self.cache[key] is not set. Since benchmark will call the kernel many times, we do not want to print out the best config every time the kernel is called.

Reply:
I agree. but the problem is that the best config is printed only for the first run, then the config is cached and the code is here is skipped. Maybe we can change line 94 to

`if key not in self.cache or self.verbose:`, then it will do the tuning and print best tuning config every time.

Reply:
> but the problem is that the best config is printed only for the first run

The best config is printed once for a given key. When the kernel is called with a different key, the tuning process will be triggered and the best config is printed, also once. 
`do_bench()` calls the kernel with the same key many times, but they all use the same best config. We do not need to print out the best config every time for the same key.